akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardingState.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed.internal
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.cluster.sharding.{ ClusterSharding, ShardRegion }
+import akka.cluster.sharding.ShardRegion.CurrentShardRegionState
+import akka.actor.typed.scaladsl.adapter._
+import akka.annotation.InternalApi
+import akka.cluster.sharding.typed.{ ClusterShardingQuery, GetShardRegionState }
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+object ShardingState {
+
+  def behavior(untypedSharding: ClusterSharding): Behavior[ClusterShardingQuery] = Behaviors.receiveMessage {
+    case GetShardRegionState(key, replyTo) ⇒
+      if (untypedSharding.getShardTypeNames.contains(key.name)) {
+        untypedSharding.shardRegion(key.name).tell(ShardRegion.GetShardRegionState, replyTo.toUntyped)
+      } else {
+        replyTo ! CurrentShardRegionState(Set.empty)
+      }
+      Behavior.same
+  }
```

Comment:
Do we really need an actor in-between for this, can't we just do an ask and map the response?

Reply:
And offer a Future based API to the users? This may get messy as we add the other queries. I only did one query to show someone how to do it so they can do the others

Reply:
Not sure what I was thinking here, you can just disregard :)